cmd/kubeless/topicCreate.go

```diff
@@ -39,52 +40,61 @@ var topicCreateCmd = &cobra.Command{
 		if err != nil {
 			logrus.Fatal(err)
 		}
-
 		topicName := args[0]
-		command := []string{"bash", "/opt/bitnami/kafka/bin/kafka-topics.sh", "--zookeeper", "zookeeper." + ctlNamespace + ":2181", "--replication-factor", "1", "--partitions", "1", "--create", "--topic", topicName}
 
-		execCommand(command, ctlNamespace)
+		conf, err := utils.BuildOutOfClusterConfig()
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		k8sClientSet := utils.GetClientOutOfCluster()
+
+		err = createTopic(conf, k8sClientSet, ctlNamespace, topicName, cmd.OutOrStdout())
+		if err != nil {
+			logrus.Fatal(err)
+		}
 	},
 }
 
+func createTopic(conf *rest.Config, clientset kubernetes.Interface, ctlNamespace, topicName string, out io.Writer) error {
+	command := []string{
+		"bash", "/opt/bitnami/kafka/bin/kafka-topics.sh",
+		"--zookeeper", "zookeeper." + ctlNamespace + ":2181",
+		"--replication-factor", "1",
+		"--partitions", "1",
+		"--create",
+		"--topic", topicName,
+	}
+	return execCommand(conf, clientset, ctlNamespace, command, out)
+}
+
 // wrapper of kubectl exec
 // execCommand executes a command to kafka pod
-func execCommand(command []string, ctlNamespace string) {
-	f := cmdutil.NewFactory(nil)
-
-	k8sClientSet := utils.GetClientOutOfCluster()
+func execCommand(conf *rest.Config, k8sClientSet kubernetes.Interface, ctlNamespace string, command []string, out io.Writer) error {
 	pods, err := utils.GetPodsByLabel(k8sClientSet, ctlNamespace, "kubeless", "kafka")
 	if err != nil {
-		logrus.Fatalf("Can't find the kafka pod: %v", err)
+		return fmt.Errorf("Can't find the kafka pod: %v", err)
 	} else if len(pods.Items) == 0 {
-		logrus.Fatalln("Can't find any kafka pod")
+		return fmt.Errorf("Can't find any kafka pod")
 	}
-	params := &k8scmd.ExecOptions{
-		StreamOptions: k8scmd.StreamOptions{
-			Namespace:     ctlNamespace,
-			PodName:       pods.Items[0].Name,
-			ContainerName: "broker",
-			In:            nil,
-			Out:           os.Stdout,
-			Err:           os.Stderr,
-			TTY:           false,
-		},
-		Executor: &k8scmd.DefaultRemoteExecutor{},
-		Command:  command,
-	}
-	config, err := f.ClientConfig()
-	if err != nil {
-		logrus.Fatalln(err)
+
+	opts := v1.PodExecOptions{
+		Stdin:     false,
+		Stdout:    true,
+		Stderr:    true,
+		Container: "broker",
+		Command:   command,
 	}
-	params.Config = config
 
-	fClientset, err := f.ClientSet()
+	cmdOut, err := utils.Exec(conf, pods.Items[0].Name, ctlNamespace, &opts)
 	if err != nil {
-		logrus.Fatalln(err)
+		return err
 	}
-	params.PodClient = fClientset.Core()
+	defer cmdOut.Close()
 
-	if err := params.Run(); err != nil {
-		logrus.Fatalln(err)
+	if _, err := io.Copy(out, cmdOut); err != nil {
```

Comment:
why are you doing this copy instead of printing to stdout? (if that is what this code does)

Reply:
Yes, this is copying to stdout.  I can't directly pass `out` to `Exec()` since despite the names, one is an `io.Writer` (aka `os.stdout`) and the other is an `io.Reader` (the connection coming from the remote pod).  Something somewhere needs to read bytes from the TCP socket and write them to stdout - and that something is here :)

Reply:
_understands_